### PR TITLE
[SPARK-15599][SQL][DOCS] API docs for `createDataset` functions in SparkSession

### PR DESCRIPTION
**Pull request by sameeragarwal**
## What changes were proposed in this pull request?

Adds API docs and usage examples for the 3 `createDataset` calls in `SparkSession`
## How was this patch tested?

N/A

### CODE REVIEW

**Comment by sameeragarwal** [comment]
cc @rxin

File: sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala [hunk 1]
```diff
@@ -376,6 +376,38 @@ class SparkSession private(
...
+   *   // |Michael| 29|
+   *   // |   Andy| 30|
+   *   // | Justin| 19|
+   *   // +-------+---+
+   * }}}
+   *
+   * @since 2.0.0
+   * @group dataset
+   */
   def createDataset[T : Encoder](data: Seq[T]): Dataset[T] = {
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala]: can you add experimental to all 3?

**Comment by sameeragarwal** [comment]
test this please

**Comment by andrewor14** [comment]
Looks great

**Comment by andrewor14** [comment]
Merging inot master 2.0

**Pull request closed without merging**